pkg/cac/cac.go

```diff
@@ -2,25 +2,70 @@ package cac
 
 import (
 	"encoding/binary"
+	"errors"
 
 	"github.com/ethersphere/bee/pkg/bmtpool"
 	"github.com/ethersphere/bee/pkg/swarm"
 )
 
+var (
+	errTooShortChunkData = errors.New("short chunk data")
+	errTooLargeChunkData = errors.New("data too large")
+)
+
+// New creates a new content address chunk by initializing a span and appending the data to it.
 func New(data []byte) (swarm.Chunk, error) {
-	hasher := bmtpool.Get()
-	defer bmtpool.Put(hasher)
+	if len(data) > swarm.ChunkSize+swarm.SpanSize {
```

Comment:
actually, here `data` can only be `0<len<=swarm.ChunkSize`, since the span is created separately below. A unit test that checks these edge cases would be good.

Reply:
I fixed it now and it is already tested in the new `TestChunkInvariants`. But currently I'm only aware of two cases:
1. Data is smaller than the spanSize
2. Data is bigger than the maxChunkSize

Do you see more cases?